enterprise/frontend/src/embedding-sdk/hooks/private/use-sdk-license-problem.ts

```diff
@@ -0,0 +1,39 @@
+import { useEffect, useMemo, useRef } from "react";
+
+import type { SDKConfig } from "embedding-sdk";
+import { getSdkLicenseProblem } from "embedding-sdk/lib/license-problem";
+import { printLicenseProblemToConsole } from "embedding-sdk/lib/print-license-problem";
+import { useSelector } from "metabase/lib/redux";
+import { getApplicationName } from "metabase/selectors/whitelabel";
+import { getTokenFeature } from "metabase/setup/selectors";
+
+export function useSdkLicenseProblem(config: SDKConfig) {
+  const { allowConsoleLog = true } = config;
+
+  const hasLoggedRef = useRef(false);
+  const appName = useSelector(getApplicationName);
+
+  const hasFeatureFlag = useSelector(state => {
```

Comment:
There is a known edge case with CORS on non-localhost origins. If the user hasn't entered their license key and haven't configured their authorized origins yet, the CORS headers will be missing from the `/api/session/properties` endpoint, therefore we can't query whether the feature flag exists. [We're addressing this.](https://metaboat.slack.com/archives/C063Q3F1HPF/p1726050178995979)